src/style-property-map.js

```diff
@@ -86,7 +86,23 @@
     if (!propertyDictionary().isSupportedProperty(property)) {
       throw new TypeError('Cannot use has method for ' + property + ' because it is not a supported CSS property');
     }
-    return !!this._styleObject[property]
+    return !!this._styleObject[property];
+  };
+
+  StylePropertyMap.prototype.getProperties = function() {
+    var specifiedPropertyIndex = 0;
+    var isSpecifiedProperty = true;
+    var propertyArray = [];
+
+    while (isSpecifiedProperty) {
```

Comment:
You're iterating through the properties in the CSSStyleDeclaration (ComputedStyle etc)  here right? If so, CSSStyleDeclaration is iterable, so it has a length and you can write

```javascript
for (var i = 0; i < this._styleObject.length; i++) {
  propertyArray.push(this._styleObject[i]);
}
```